app/common/src/map/PriceMarker.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+
+import * as React from 'react';
+import { StyleSheet, View, Text } from 'react-native';
+import type { CurrencyInfo } from '../../../hotels/src/types/datastructures';
+
+const styles = StyleSheet.create({
+  container: {},
+  bubble: {
+    flexDirection: 'row',
+    paddingVertical: 5,
+    paddingHorizontal: 8,
+    backgroundColor: '#fff',
+    borderRadius: 3,
+    borderColor: '#fff',
+    borderWidth: 2,
+  },
+  bubbleSelected: {
+    backgroundColor: '#4da2ab',
+  },
+  price: {
+    color: '#000',
+    fontSize: 18,
+  },
+  priceSelected: {
+    color: '#fff',
+  },
+  currency: {
+    color: 'rgba(0, 0, 0, 0.5)',
+    fontSize: 15,
+  },
+  currencySelected: {
+    color: '#fff',
+  },
+  arrowBorder: {
+    backgroundColor: 'transparent',
+    borderColor: 'transparent',
+    borderWidth: 8,
+    borderTopColor: '#ffffff',
+    alignSelf: 'center',
+  },
+  arrow: {
+    backgroundColor: 'transparent',
+    borderColor: 'transparent',
+    borderWidth: 7,
+    borderTopColor: '#fff',
+    alignSelf: 'center',
+    marginTop: -18,
+  },
+  arrowSelected: {
+    borderTopColor: '#4da2ab',
+  },
+});
+
+type Props = {|
+  isSelected?: boolean,
+  price: number,
+  currencyInfo: CurrencyInfo,
+|};
+
+const renderCurrency = (props: Props) => (
+  <Text style={[styles.currency, props.isSelected && styles.currencySelected]}>
+    {props.currencyInfo.currency}
+  </Text>
+);
+
+const PriceMarker = (props: Props) => {
+  const { isSelected } = props;
+
+  return (
+    <View style={styles.container}>
+      <View style={[styles.bubble, isSelected && styles.bubbleSelected]}>
+        {props.currencyInfo.position === 'before' && renderCurrency(props)}
+        <Text style={[styles.price, isSelected && styles.priceSelected]}>
+          {props.price}
+        </Text>
+        {props.currencyInfo.position === 'after' && renderCurrency(props)}
+      </View>
+      <View style={[styles.arrowBorder]} />
+      <View style={[styles.arrow, isSelected && styles.arrowSelected]} />
```

Comment:
sounds good, but it smells by code duplication (see arrow & arrowSelected for example)

Reply:
I am pretty sure you can figure out how to do it without duplication. Try it and if it's not going to be more readable, we can discard my comment... :)

Reply:
I read this again and let me respond differently: Why code duplication? This is not related to the `isSelected` condition. One component for the pure marker (`MapMarker`) with all the styles and second component with additional content (`PriceMapMarker`). So the second component will basically be only:

```js
render = () => <MapMarker>{props.price} {props.currency}</MapMarker>
```

The point is to separate the concerns because 1) we may easily reuse the marker itself and 2) we may have quite a lot of logic here (before, after currency, ...). But since we are no longer dealing with currency formatting, I am ok to leave it now "as is" until we decide to improve it. I just wanted to make my previous comment more clear... 🙂